tools/enumerate-devices/rs-enumerate-devices.cpp

```diff
@@ -386,25 +403,49 @@ int main(int argc, char** argv) try
 
         if (show_modes)
         {
-            for (auto&& sensor : dev.query_sensors())
+            size_t w_res = 12;
+            size_t w_fps = 10;
+            size_t w_format = 10;
+
+            for( auto&& sensor : dev.query_sensors() )
             {
-                cout << "Stream Profiles supported by " << sensor.get_info(RS2_CAMERA_INFO_NAME) << endl;
+                cout << "Stream Profiles supported by " << sensor.get_info( RS2_CAMERA_INFO_NAME ) << endl;
+                cout << " Supported modes:\n";
 
-                cout << " Supported modes:\n" << setw(16) << "    stream" << setw(16)
-                    << " resolution" << setw(10) << " fps" << setw(10) << " format" << endl;
+                size_t w_stream = 10;
+                bool video_stream = false;
+                for( auto&& profile : sensor.get_stream_profiles() )
+                {
+                    w_stream = std::max( profile.stream_name().length(), w_stream );
+                    if( auto video = profile.as<video_stream_profile>() )
+                        video_stream = true;
+                }
+                w_stream += 2;
+
+                // Heading
+                cout << "    ";
+                if( verbose )
+                    cout << "       ";  // (sid,index)
+                cout << setw( w_stream ) << "STREAM";
+                if( video_stream )
+                    cout << setw( w_res ) << "RESOLUTION";
```

Comment:
In this code section we set width and add a const string, Where is the data itself?

Reply:
The string is the data